libs/mutantschemata/source/mutantschemata/externals.d

```diff
@@ -0,0 +1,81 @@
+/**
+Copyright: Copyright (c) 2019, Niklas Pettersson. All rights reserved.
+License: MPL-2
+Author: Niklas Pettersson (nikpe353@student.liu.se)
+
+This Source Code Form is subject to the terms of the Mozilla Public License,
+v.2.0. If a copy of the MPL was not distributed with this file, You can obtain
+one at http://mozilla.org/MPL/2.0/.
+
+All the external items used in the api between C++ and D code
+*/
+module mutantschemata.externals;
+
+// External C++ string implementation
+extern (C++, CppString) {
+    extern (C++) struct CppBytes {
+        void* ptr;
+        int length;
+
+        void destroy();
+    }
+    extern (C++) struct CppStr {
+        void* cppStr;
+
+        const(void)* ptr();
+        int length();
+        void destroy();
+        void put(char);
+    }
+    extern (C++) CppBytes getBytes();
```

Comment:
It is possible to write `extern(C++):`. Then everything statement after it is tagged as being `c++`.